exporter/exporterhelper/internal/queue_capacity.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+
+import (
+	"sync/atomic"
+)
+
+type itemsCounter interface {
+	ItemsCount() int
+}
+
+// QueueCapacityLimiter is an interface to control the capacity of the queue.
+type QueueCapacityLimiter[T any] interface {
```

Comment:
Instead of this complicated interface, what if we have only a "Sizer":

```golang

// Can even be just a typedef of the func.
type Sizer[T any interface {
	sizeof(el T) uint64
}

type baseCapacityLimiter[T any] struct {
	used *atomic.Uint64
	cap  uint64
	sizer Sizer[T]
}

func newBaseCapacityLimiter[T any](capacity int, sizer Sizer[T]) baseCapacityLimiter[T] {
	return baseCapacityLimiter[T]{
		used: &atomic.Uint64{},
		cap:  uint64(capacity),
		sizer: sizer,
	}
}

func (bcl *baseCapacityLimiter[T]) Capacity() int {
	return int(bcl.cap)
}

func (bcl *baseCapacityLimiter[T]) Size() int {
	return int(bcl.used.Load())
}

// nolint:unused // false positive https://github.com/dominikh/go-tools/issues/1440
func (bcl *baseCapacityLimiter[T]) claim(el T) bool {
	size := sizer.sizeof(el)
	if bcl.used.Add(size) > bcl.cap {
		bcl.used.Add(^(size - 1))
		return false
	}
	return true
}

// nolint:unused // false positive https://github.com/dominikh/go-tools/issues/1440
func (bcl *baseCapacityLimiter[T]) release(el T) {
	size := sizer.sizeof(el)
	bcl.used.Add(^(size - 1))
}

type requestsSizer[T any] struct {}

func NewRequestsCapacityLimiter[T any](capacity int) QueueCapacityLimiter[T] {
	return &requestsSizer[T]{}
}

// nolint:unused // false positive https://github.com/dominikh/go-tools/issues/1440
func (rcl *requestsSizer[T]) sizeOf(T) uint64 {
	return 1
}

```

Reply:
Thanks for the suggestion. It helped to get rid of the interface and `nolint` comments